main/boofcv-ip/src/main/java/boofcv/alg/color/ColorDiff.java

```diff
@@ -0,0 +1,240 @@
+    }
+
+
+    /**
+     * CIE94 color difference equation.
+     * This is the an extended CIE76 to address perceptual non-uniformities in CIELAB color space.
+     *
+     * @param lab1 color1
+     * @param lab2 color2
+     * @param KL depend on application: graphics arts = 1, textiles = 2
+     * @param KC usually = 1
+     * @param KH usually = 1
+     * @param K1 depend on application: graphics arts = 0.045, textiles = 0.048
+     * @param K2 depend on application: graphics arts = 0.015, textiles = 0.014
+     * @return colors distance
+     */
+    public static double deltaECIE94(double[] lab1, double [] lab2, double KL, double KC, double KH,
+                                     double K1, double K2) {
+        return (deltaECIE94(lab1[l], lab1[a], lab1[b], lab2[l], lab2[a], lab2[b], KL, KC, KH, K1, K2 ));
+    }
+
+    public static double deltaECIE94(double l1, double a1, double b1, double l2, double a2, double b2,
+                                     double KL, double KC, double KH, double K1, double K2) {
+        double deltaL = l1 - l2;
+        double deltaA = a1 - a2;
+        double deltaB = b1 - b2;
+        double C1 = Math.sqrt(Math.pow(a1,2) + Math.pow(b1,2));
+        double C2 = Math.sqrt(Math.pow(a2,2) + Math.pow(b2,2));
+        double deltaC = C1 - C2;
+        // Changed to avoid NaN due limited arithmetic precision
+        //double deltaH = Math.sqrt(Math.pow(deltaA,2) + Math.pow(deltaB,2) - Math.pow(deltaC,2));
+        double deltaH = Math.pow(deltaA,2) + Math.pow(deltaB,2) - Math.pow(deltaC,2);
+        double SL = 1;
+        double SC = 1 + K1*C1;
+        double SH = 1 + K2*C1;
+        // Changed to avoid NaN due limited arithmetic precision
+        //return (Math.sqrt(Math.pow(deltaL / (KL * SL),2) + Math.pow(deltaC / (KC * SC),2) + Math.pow(deltaH / (KH * SH),2)));
+        return (Math.sqrt(Math.pow(deltaL / (KL * SL),2) + Math.pow(deltaC / (KC * SC),2) + deltaH * Math.pow( 1 / (KH * SH),2)));
+    }
+
+
+        /**
+		 * CIE94 color difference equation.
+		 * CIE94 with reasonably parameters (graphics arts). In doubt, use this one!!
+		 *
+		 * @param lab1 color1
+		 * @param lab2 color2
+		 * @return colors distance
+		 */
+    public static double deltaECIE94(double[] lab1, double [] lab2) {
+        return(deltaECIE94(lab1[l], lab1[a], lab1[b], lab2[l], lab2[a], lab2[b]));
+    }
+
+    public static double deltaECIE94(double l1, double a1, double b1, double l2, double a2, double b2) {
+        return(deltaECIE94(l1, a1, b1, l2, a2, b2,1,1,1,0.045,0.015));
+    }
+
+
+    /**
+     * CIEDE2000 color difference equation.
+     * Super crazy color difference equation. Extends the CIE94 to a very nerdy level. Should be better than CIE94,
+     * but more expensive and should make no difference in practical applications according
+     * to [Color Apperance Models, Mark D Fairchild]
+     *
+     * @param lab1 color1
+     * @param lab2 color2
+     * @return color distance
```

Comment:
distance is >= 0?

Reply:
Always positive. Zero is good color matching, big number is big mismatch. I'll make a note.
Scales changes with the method used.